### PR TITLE
[ANT-2206] Full exemple

### PR DESCRIPTION
**Pull request by JasonMarechal25**
Ful exemple of the parsing + conversion to model of a library

### CODE REVIEW

File: src/solver/modelParser/encoders.hxx [hunk 1]
```diff
@@ -54,15 +61,15 @@ struct convert<Antares::Solver::ModelParser::ValueType>
         {
             return false;
         }
-        if (node.as<std::string>() == "FLOAT")
+        if (node.as<std::string>() == "continuous")
```
Comment (flomnes) [inline src/solver/modelParser/encoders.hxx]: you don't need to convert to `std::string` N times, you can convert it once and for all (l. 64, l. 68, l. 72). The compiler may optimize this away, not sure though

File: src/tests/src/solver/modelParser/enum_operators.h [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+namespace Antares::Solver::ObjectModel
+{
+inline std::ostream& operator<<(std::ostream& os, const ValueType& value_type)
+{
+    using namespace std::string_literals;
+    switch (value_type)
+    {
+    case ValueType::FLOAT:
+        os << "FLOAT"s;
```
Comment (flomnes) [inline src/tests/src/solver/modelParser/enum_operators.h]:
```suggestion
        os << "float"s;
```

File: src/tests/src/solver/modelParser/enum_operators.h [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        break;
+    case ValueType::INTEGER:
+        os << "INTEGER"s;
```
Comment (flomnes) [inline src/tests/src/solver/modelParser/enum_operators.h]:
```suggestion
        os << "integer"s;
```

File: src/tests/src/solver/modelParser/enum_operators.h [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        break;
+    case ValueType::BOOL:
+        os << "BOOL"s;
```
Comment (flomnes) [inline src/tests/src/solver/modelParser/enum_operators.h]:
```suggestion
        os << "boolean"s;
```

File: src/tests/src/solver/modelParser/testModelParser.cpp [hunk 5]
```diff
@@ -230,7 +232,7 @@ BOOST_AUTO_TEST_CASE(model_can_contain_multiple_parameters)
                   description: "model_description"
                   parameters:
                       - id: "param_name1"
-                        time-dependent: false
+                        time-dependent: FALSE
                         scenario-dependent: false
```
Comment (flomnes) [inline src/tests/src/solver/modelParser/testModelParser.cpp:236]: I find it odd to mix FALSE, false, TRUE, true

**Review by flomnes — COMMENTED** [review]
Please harmonize true/TRUE, FLOAT/continuous, etc.

File: src/tests/src/solver/modelParser/testModelParser.cpp (hunk 5 above)
Comment (pet-mit) [reply src/tests/src/solver/modelParser/testModelParser.cpp:236]: It seems to me that he is just making use of and testing all the possibilities offered by the yml parser by default. IMO we can keep them for now, and if we need to limit the possibilities, do it in another PR ?

Comment (JasonMarechal25) [reply src/tests/src/solver/modelParser/testModelParser.cpp:236]: That's it

**Review by pet-mit — APPROVED** [review]
OK for me if it's of with @flomnes

File: src/tests/src/solver/modelParser/test_full.cpp [hunk 6]
```diff
@@ -0,0 +1,510 @@
...
+using namespace std::string_literals;
+using namespace Antares::Solver;
+
+void checkParameter(const ObjectModel::Parameter& parameter,
+                    const std::string& name,
+                    bool timeDependent,
+                    bool scenarioDependent,
+                    ObjectModel::ValueType type)
+{
+    std::cout << "Parameter: " << parameter.Id() << std::endl;
```
Comment (flomnes) [inline src/tests/src/solver/modelParser/test_full.cpp:45]:
```suggestion
```

File: src/tests/src/solver/modelParser/test_full.cpp [hunk 7]
```diff
@@ -0,0 +1,510 @@
...
+    BOOST_CHECK_EQUAL(parameter.Type(), type);
+}
+
+void checkVariable(const ObjectModel::Variable& variable,
+                   const std::string& name,
+                   const std::string& lowerBound,
+                   const std::string& upperBound,
+                   ObjectModel::ValueType type)
+{
+    std::cout << "Variable: " << variable.Id() << std::endl;
```
Comment (flomnes) [inline src/tests/src/solver/modelParser/test_full.cpp:58]:
```suggestion
```

File: src/tests/src/solver/modelParser/test_full.cpp [hunk 8]
```diff
@@ -0,0 +1,510 @@
...
+    BOOST_CHECK_EQUAL(variable.LowerBound().Value(), lowerBound);
+    BOOST_CHECK_EQUAL(variable.UpperBound().Value(), upperBound);
+    BOOST_CHECK_EQUAL(variable.Type(), type);
+}
+
+void checkConstraint(const ObjectModel::Constraint& constraint,
+                     const std::string& name,
+                     const std::string& expression)
+{
+    std::cout << "Constraint: " << constraint.Id() << std::endl;
```
Comment (flomnes) [inline src/tests/src/solver/modelParser/test_full.cpp:69]:
```suggestion
```

File: src/tests/src/solver/modelParser/enum_operators.h [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#pragma once
+#include <iostream>
```
Comment (flomnes) [inline src/tests/src/solver/modelParser/enum_operators.h]:
```suggestion
#include <ostream>
```

File: src/solver/modelParser/encoders.hxx [hunk 10]
```diff
@@ -54,15 +61,16 @@ struct convert<Antares::Solver::ModelParser::ValueType>
         {
             return false;
         }
-        if (node.as<std::string>() == "FLOAT")
+        auto value = node.as<std::string>();
```
Comment (flomnes) [inline src/solver/modelParser/encoders.hxx]:
```suggestion
        const auto value = node.as<std::string>();
```

File: src/solver/modelParser/encoders.hxx [hunk 11]
```diff
@@ -29,6 +29,13 @@
 // Implement convert specializations
 namespace YAML
 {
+
+template<typename T>
+inline T as_fallback_default(const Node& n)
+{
+    return n.as<T>(T());
```
Comment (flomnes) [inline src/solver/modelParser/encoders.hxx:46]: Can you add a comment to explain why we need this ?

**Review by flomnes — CHANGES_REQUESTED** [review]
Minor remarks

File: src/tests/src/solver/modelParser/test_full.cpp (hunk 6 above)
Comment (JasonMarechal25) [reply src/tests/src/solver/modelParser/test_full.cpp:45]: Helpful when an error occurs. Else we don't really know what failed.

**Review by flomnes — APPROVED** [review]

**Pull request merged**